xbmc/pvr/PVRDatabase.cpp

```diff
@@ -636,6 +636,18 @@ bool CPVRDatabase::Persist(CPVRChannelGroup &group)
   {
     CSingleLock lock(group.m_critSection);
 
+    /* Prevent AutoIncValue usage for internal groups as long they have fixed value */
+    if (group.GroupID() <= 0 && group.IsInternalGroup())
+      group.m_iGroupId = group.IsRadio() ?PVR_INTERNAL_GROUP_ID_RADIO:PVR_INTERNAL_GROUP_ID_TV;
+
+    /* First insert the # internal groups */
+    if(m_pDS->lastinsertid()==0)
+    {
+      strQuery = PrepareSQL("INSERT INTO channelgroups (idGroup) VALUES (NULL)");
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_RADIO
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_TV
```

Comment:
something like that yes. but please create a separate method e.g. `CPVRDatabase::InsertInternalGroup(bool bRadio)` and do the necessary check and insert it (if exists we do nothing). Call this method in `CPVRChannelGroups::Load()` before `// load groups from the database` after we push the internal group to `m_groups`.